src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,113 @@
-public class StringManipulation implements StringManipulationInterface {
+/* Nadav Horowitz 6/3/2023 StringManipulation.java
+ * This file contains various methods for manipulation of a String object.
+ * The methods implemented include count, removeNthCharacter, getSubStrings, and restoreString.
+ */
+import java.util.Arrays;
 
+public class StringManipulation implements StringManipulationInterface {
+	
+	public String string;
+	
+	
     @Override
+    //Getter method for String field
     public String getString() {
-        return null;
+        return this.string;
     }
 
+    
     @Override
+    //Setter method for String field
     public void setString(String string) {
+    	this.string = string;
     }
 
+    
     @Override
+    //Returns number of words present in String object
     public int count() {
-        return 0;
+    	//null case and empty case
+    	if (string == null || string.length() == 0)
+    		return 0;
+    	//trim leading and ending whitespace
+    	String copy = this.getString();
```

Comment:
Since the _string_ is already a member of the class, there's no need to call a getter method of getString() to access it, nor is there any reason to copy the string every time -- none of these methods are mutators.

Reply:
I make a copy here because afterwards I call the String.trim() method, which removes leading and lagging whitespace. Just for the sake of not changing the original string.